src/main/java/mekanism/common/content/teleporter/TeleporterFrequency.java

```diff
@@ -57,4 +71,28 @@ public Coord4D getClosestCoords(Coord4D coord) {
         }
         return closest;
     }
+
+    @Override
+    protected void read(CompoundNBT nbtTags) {
+        super.read(nbtTags);
+        NBTUtils.setEnumIfPresent(nbtTags, NBTConstants.COLOR, EnumColor::byIndexStatic, (value) -> color = value);
+    }
+
+    @Override
+    protected void read(PacketBuffer dataStream) {
+        super.read(dataStream);
+        color = EnumColor.byIndexStatic(dataStream.readVarInt());
```

Comment:
I believe this should be able to be `color = dataStream.readEnum(EnumColor.class);`

Reply:
I can change it to that, but i just stole this from the qio.

Reply:
done.